src/os/bluestore/BlueStore.cc

```diff
@@ -7121,6 +7121,7 @@ int BlueStore::getattrs(
 
 int BlueStore::list_collections(vector<coll_t>& ls)
 {
+  ls.reserve(coll_map.size());
   RWLock::RLocker l(coll_lock);
```

Comment:
can you move this down a line under the coll_lock?  It's not safe to call size() outside of the lock.

Reply:
alloc memory may need long time and i don't hope by coll_lock. And if size() increase is small conceptual event. We don't consider this.